src/main/scala/com/campudus/tableaux/database/domain/column.scala

```diff
@@ -136,63 +227,178 @@ object SimpleValueColumn {
+    value match {
+      case v: java.lang.Boolean => v
+      case v: Boolean => v
+    }
+  }
+}
 
-case class CurrencyColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[Number](CurrencyType)(languageType)
+case class DateColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[String](DateType)(languageType) {
+  override def checkValidSingleValue[B](value: B): Try[String] = Try {
+    value match {
+      case v: String => v
+    }
 
-case class BooleanColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[Boolean](BooleanType)(languageType)
+    /*value match {
+      case v: String => LocalDate.parse(v)
+      case v: LocalDate => v
+    }*/
+  }
+}
 
-case class DateColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[LocalDate](DateType)(languageType)
+case class DateTimeColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[String](DateTimeType)(languageType) {
+  override def checkValidSingleValue[B](value: B): Try[String] = Try {
+    value match {
+      case v: String => v
+    }
 
-case class DateTimeColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[DateTime](DateTimeType)(languageType)
+    /*value match {
+      case v: String => DateTime.parse(v)
+      case v: DateTime => v
+    }*/
+  }
+}
 
 /*
  * Special column types
  */
-case class LinkColumn(override val columnInformation: ColumnInformation, to: ColumnType[_], linkId: LinkId, linkDirection: LinkDirection) extends ColumnType[JsonArray] {
+case class LinkColumn(override val columnInformation: ColumnInformation, to: ColumnType[_], linkId: LinkId, linkDirection: LinkDirection) extends ColumnType[Seq[RowId]] with LazyLogging {
   override val kind = LinkType
   override val languageType = to.languageType
 
   override def getJson: JsonObject = super.getJson mergeIn Json.obj("toTable" -> to.table.id, "toColumn" -> to.getJson)
+
+  override def checkValidValue[B](value: B): Try[Option[Seq[RowId]]] = Try {
+    val castedValue = value match {
+      case x: Long =>
+        Seq(x)
+
+      case x: Int =>
+        Seq(x.toLong)
+
+      case x: Seq[_] =>
+        x.map {
+          case id: RowId => id
+          case obj: JsonObject => obj.getLong("id").toLong
+        }
+
+      case x: JsonObject if x.containsKey("to") =>
+        import ArgumentChecker._
+        hasLong("to", x) match {
+          case arg: OkArg[Long] =>
+            Seq(arg.get)
+          case _ =>
+            throw InvalidJsonException(s"A link column expects a JSON object with to values, but got $x", "link-value")
+        }
```

Comment:
This looks like we should think about adding a `getOrElse` or a `toOption`/`toTry` method to our `OkArg`